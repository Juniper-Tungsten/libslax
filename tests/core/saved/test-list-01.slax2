version 1.1;

ns my = "http://my.example.com";

var $key = 1;
var $new = {
    trace {
        if ($key) {
            <trace-test> "testing...1...2...3...";
        
        } else {
            <trace-test> "testing...not...";
        }
    }
}
var $min = 1;
var $max = {
    trace "min: " _ $min;
    expr 10;
}

match /list {
    <top> {
        trace my:test(10);
        trace "test";
        trace "count" _ count(.);
        trace $max - $min;
        trace {
            <top> {
                if ($min == 0) {
                    expr "bad";
                
                } else {
                    expr "good";
                }
            }
        }
        
        for-each (item) {
            trace name;
            trace "item: " _ name;
            <found> name;
        }
    }
}

function my:test ($value) {
    result {
        <test> $value;
    }
}
