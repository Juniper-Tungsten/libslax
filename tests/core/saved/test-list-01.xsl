<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:my="http://my.example.com" xmlns:slax="http://xml.libslax.org/slax" xmlns:slax-func="http://exslt.org/functions" version="1.0" extension-element-prefixes="slax slax-func">
  <xsl:variable name="key" select="1"/>
  <xsl:variable name="new">
    <slax:trace xmlns:slax="http://xml.libslax.org/slax">
      <xsl:choose>
        <xsl:when test="$key">
          <trace-test>testing...1...2...3...</trace-test>
        </xsl:when>
        <xsl:otherwise>
          <trace-test>testing...not...</trace-test>
        </xsl:otherwise>
      </xsl:choose>
    </slax:trace>
  </xsl:variable>
  <xsl:variable name="min" select="1"/>
  <xsl:variable name="max">
    <slax:trace xmlns:slax="http://xml.libslax.org/slax" select="concat(&quot;min: &quot;, $min)"/>
    <xsl:value-of select="10"/>
  </xsl:variable>
  <xsl:template match="/list">
    <top>
      <slax:trace xmlns:slax="http://xml.libslax.org/slax" select="my:test(10)"/>
      <slax:trace xmlns:slax="http://xml.libslax.org/slax" select="&quot;test&quot;"/>
      <slax:trace xmlns:slax="http://xml.libslax.org/slax" select="concat(&quot;count&quot;, count(.))"/>
      <slax:trace xmlns:slax="http://xml.libslax.org/slax" select="$max - $min"/>
      <slax:trace xmlns:slax="http://xml.libslax.org/slax">
        <top>
          <xsl:choose>
            <xsl:when test="$min = 0">
              <xsl:text>bad</xsl:text>
            </xsl:when>
            <xsl:otherwise>
              <xsl:text>good</xsl:text>
            </xsl:otherwise>
          </xsl:choose>
        </top>
      </slax:trace>
      <xsl:for-each select="item">
        <slax:trace xmlns:slax="http://xml.libslax.org/slax" select="name"/>
        <slax:trace xmlns:slax="http://xml.libslax.org/slax" select="concat(&quot;item: &quot;, name)"/>
        <found>
          <xsl:value-of select="name"/>
        </found>
      </xsl:for-each>
    </top>
  </xsl:template>
  <slax-func:function xmlns:slax-func="http://exslt.org/functions" name="my:test">
    <xsl:param name="value"/>
    <slax-func:result xmlns:slax-func="http://exslt.org/functions">
      <test>
        <xsl:value-of select="$value"/>
      </test>
    </slax-func:result>
  </slax-func:function>
</xsl:stylesheet>
