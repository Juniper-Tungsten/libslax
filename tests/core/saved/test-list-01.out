<?xml version="1.0"?>
<top xmlns:my="http://my.example.com">
  <found>toy</found>
  <found>doll</found>
</top>
